draft-irtf-cfrg-hpke.md

```diff
@@ -1567,32 +1611,62 @@ HPKE is not tolerant of lost messages. Applications MUST be able to detect when
+
+HPKE assumes that the sender and recipient agree on what algorithms to use.
+Depending on how these algorithms are negotiated, it may be possible for an
+intermediary to force the two parties to use suboptimal algorithms.
+
+### Replay Protection
+
+The requirement that ciphertexts be presented to the `ContextR.Open()` function
+in the same order they were generated by `ContextS.Seal()` provides a degree of
+replay protection within a stream of ciphertexts resulting from a given context.
+HPKE provides no other replay protection.
+
+### Forward Secrecy
 
-HPKE does not provide several features that a more high-level protocol might
-provide, for example:
+HPKE ciphertexts are not forward secret with respect to recipient compromise
+in any mode. This is because no ephemeral randomness but only long-term secrets
+are used on the side of the recipient.
 
-* Downgrade prevention - HPKE assumes that the sender and recipient agree on
-  what algorithms to use.  Depending on how these algorithms are negotiated, it
-  may be possible for an intermediary to force the two parties to use
-  suboptimal algorithms.
+HPKE ciphertexts are forward secret with respect to sender compromise in all
+modes. This is because ephemeral randomness is used on the sender's side, which
+is supposed to be erased directly after computation of the KEM shared secret and
+ciphertext.
 
-* Replay protection - The requirement that ciphertexts be presented to the
-  `ContextR.Open()` function in the same order they were generated by `ContextS.Seal()`
-  provides a degree of replay protection within a stream of ciphertexts
-  resulting from a given context.  HPKE provides no other replay protection.
+For DHKEM, HPKE is morally similar to specific one-way Noise handshake patterns,
+which can be helpful to understand forward security and other properties {{Noise}}:
+Base mode is similar to Noise N, PSK mode to Npsk1, Auth mode to K, AuthPSK to Kpsk1.
 
-* Forward secrecy - HPKE ciphertexts are not forward-secure. In the Base and
-  Auth modes, a given ciphertext can be decrypted if the recipient's public
-  encryption key is compromised. In the PSK and AuthPSK modes, a given
-  ciphertext can be decrypted if the recipient's private key and the
-  PSK are compromised.
+### Bad Ephemeral Randomness
+
+If the randomness used for KEM encapsulation is bad, i.e. of low entropy or
+compromised because of a broken or subverted random number generator, the
+confidentiality guarantees of HPKE degrade significantly. In Base mode,
+can be lost completely; in the other modes, at least forward secrecy with
+respect to sender compromise can be lost completely.
+
+Such a situation could also lead to the reuse of the same KEM shared secret
+and thus to the reuse of same (key, nonce) pairs for the AEAD.
+The AEADs specified in this document keep no or only limited security guarantees
+in case of nonce reuse. This is another attack vector against confidentiality
+in case of bad ephemeral randomness.
```

Comment:
```suggestion
in case of nonce reuse.
```

Reply:
I tried to rephrase to make clearer what I wanted to express: when nonce reuse is relevant. It's not relevant in Base mode, because there, everything is lost anyway.